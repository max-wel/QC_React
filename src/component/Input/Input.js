import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Input = props => {
  const { type, value, placeholder, label, name, onChange, errors } = props;
  const classes = cx('input is-medium is-hovered', {
    'is-danger': errors[name]
  });
  return (
    <div className="field">
      <label className="label">{label}</label>
      <div className="control">
        <input
          className={classes}
          type={type}
          placeholder={placeholder}
          value={value}
          name={name}
          onChange={onChange}
        />
      </div>
      {errors[name] && <span className="help is-danger">{errors[name]}</span>}
    </div>
  );
};

Input.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  label: PropTypes.string.isRequired
};

export default Input;
