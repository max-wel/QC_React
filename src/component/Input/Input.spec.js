import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Input from './Input';

describe('<Input /> tests', () => {
  it('renders an <Input /> component', () => {
    const props = {
      type: '',
      placeholder: '',
      label: '',
      name: '',
      onChange: jest.fn(),
      errors: {}
    };
    const output = renderer.create(<Input {...props} />);
    expect(output).toMatchSnapshot();
  });

  it('renders error text on input error', () => {
    const props = {
      name: 'email',
      errors: {
        email: 'Error'
      }
    };
    const wrapper = shallow(<Input {...props} />);
    expect(wrapper.contains('Error')).toBeTruthy();
  });
});
