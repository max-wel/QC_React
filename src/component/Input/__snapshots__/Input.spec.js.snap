// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> tests renders an <Input /> component 1`] = `
<div
  className="field"
>
  <label
    className="label"
  >
    
  </label>
  <div
    className="control"
  >
    <input
      className="input is-medium is-hovered"
      name=""
      onChange={[MockFunction]}
      placeholder=""
      required={false}
      type=""
    />
  </div>
</div>
`;
